internal/charmstore/store.go

```diff
@@ -1425,6 +1425,35 @@ func (store *Store) Search(sp SearchParams) (SearchResult, error) {
 	return result, nil
 }
 
+// List lists the store for the given ListParams.
+// It returns a ListResult containing the results of the list.
+func (store *Store) List(sp ListParams) (ListResult, error) {
+	query := store.DB.Entities().Find(sp.Filters)
+	if sp.sort == nil {
+		query = query.Sort("_id")
```

Comment:
by default I think that promulgated ones should come first. not sure if you can do that easily though.